src/pybind/rados.py

```diff
@@ -1984,6 +1984,23 @@ def lookup_snap(self, snap_name):
             raise make_ex(ret, "Failed to lookup snap %s" % snap_name)
         return Snap(self, snap_name, snap_id)
 
+    @requires(('oid', str_type), ('snap', Snap))
+    def snap_rollback(self, oid, snap):
+        """
+        Rollback an object to a snapshot
+
+        :param oid: the name of the object
+        :type snap: Snap object
+
+        :raises: :class:`TypeError`
+        :raises: :class:`Error`
+        """
+        self.require_ioctx_open()
+        ret = run_in_thread(self.librados.rados_ioctx_snap_rollback,
+                            (self.io, cstr(oid), snap.name))
```

Comment:
snap.name is populated as a python string, so we should encode it as a byte string via cstr(snap.name) when passing it to the c function here